### PR TITLE
Switch to 2DVar for snow DA + use JCB

### LINKED ISSUE #1324: Switch to 2DVar for snow DA
Move from LETKF-OI with a fake ensemble to 2DVar

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
Closes #1324 

This PR, combined with a companion workflow PR, switches the snow analysis to be 2DVar for both deterministic and ensemble mean, to use JCB to generate YAMLs and to leverage the g-w Jedi class. It also updates the IMS snow cover processing hash to use `snodl` instead of `snwpdh`

### CODE REVIEW

**Comment by CoryMartin-NOAA** [comment]
This is in draft until https://github.com/NOAA-EMC/jcb-gdas/pull/52 is merged and the hash is updated

File: parm/snow/jcb-base.yaml.j2 [hunk 1]
```diff
@@ -23,60 +21,52 @@ window_length: "{{ SNOW_WINDOW_LENGTH }}"
...
+snow_npz_anl: {{ npz_ges | default(127, true) }}
+
+snow_fv3jedi_files_path: ./fv3jedi  # Ideally this would be {{DATA}}/fv3jedi but FMS
+snow_orog_files_path: "{{ FIXgfs }}/orog/{{ CASE }}"
+snow_orog_prefix: "{{ CASE }}.mx{{ OCNRES }}"
 
 # Background
-background_path: ./bkg
+snow_background_path: {{snow_bkg_path}}
 background_ensemble_path: ./ens/mem%mem%
```
Comment (jiaruidong2017) [inline parm/snow/jcb-base.yaml.j2:59]: Does this require to provide a full path?

**Comment by CoryMartin-NOAA** [comment]
hash is updated, this is ready for review

File: parm/snow/letkfoi/ens_apply_incr_nml.j2
Comment (CoryMartin-NOAA) [inline parm/snow/letkfoi/ens_apply_incr_nml.j2:1]: note to @CoryMartin-NOAA I shoud rename this directory...

**Review by jiaruidong2017 — APPROVED** [review]
looks good to me. Just a minor question.

File: parm/snow/jcb-base.yaml.j2 (hunk 1 above)
Comment (CoryMartin-NOAA) [reply parm/snow/jcb-base.yaml.j2:59]: no it's relative to the runtime directory, this is often safer as it doesn't lead to very long filenames

File: parm/snow/jcb-prototype_2dvar.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Algorithm
+# ---------
+algorithm: 3dvar
```
Comment (YoulongXia-NOAA) [inline parm/snow/jcb-prototype_2dvar.yaml.j2:3]: 2dvar or 3dvar? your script name is 2dvar but I am not sure.

**Review by YoulongXia-NOAA — COMMENTED** [review]
Everything looks good. I am not sure that you are using 2dvar or 3dvar?

File: parm/snow/jcb-prototype_2dvar.yaml.j2 (hunk 2 above)
Comment (CoryMartin-NOAA) [reply parm/snow/jcb-prototype_2dvar.yaml.j2:3]: there is no 2dvar algorithm in JEDI, it's 3DVar just with a single value in z

File: parm/snow/jcb-base.yaml.j2 (hunk 1 above)
Comment (YoulongXia-NOAA) [reply parm/snow/jcb-base.yaml.j2:59]: Okay, that is good. I understand now. Thank you.

**Pull request merged**